Ruby/as1_ruby.rb

```diff
@@ -0,0 +1,97 @@
+#start of Program
+puts "enter the size of the grid"
+$size =  gets.chomp.to_i
+
+#Initializing Array
+$array = Array.new($size){Array.new($size , 0)}
+
+#sample Data
+$array[0][0] = 10
+$array[1][1] = 20
+$array[2][2] = 30
+$array[3][3] = 40
+$array[4][4] = 50
+
+# Hash to find the index
+$column_hash = {
+  A: 0,
+  B: 1,
+  C: 2,
+  D: 3,
+  E: 4,
+  F: 5,
+  G: 6,
+  H: 7,
+  I: 8,
+  J: 9
+}
+$expression_hash = Hash.new
+$dependent_hash = Hash.new
+# Function to print the array
+def print_array
+  for i in 0..$size-1 do
+    print "|"
+    for j in 0..$size-1 do
+      print "#{$array[i][j]} \t |"
+    end
+    print "\n"
+  end
+end
+def set_dependency (lhs , rhs)
+  cells = rhs.split(" ")
+  #temp variable to push the dependent cell into the Hash
+  temp = $dependent_hash[cells[0].to_sym]||[]
```

Comment:
Give one space gap b/w variables and operator.
$dependent_hash[cells[0].to_sym] || [ ]